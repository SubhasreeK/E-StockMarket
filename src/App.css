.App {
  text-align: center;
}
.button{
  margin-top: 30px;
  
}
.leftnavlink{
  margin-left: 340px;

}
.navlinks{
  margin-left: 40px;

}
.login{
  height: 500px;
}
.responsiveImg{
  width:100%;
   height:100%;
   object-fit:cover;
   image-rendering: smooth;
   object-position:50% 50%;
}
.navstyle{
  margin-left: 1rem;
}
.formvalidationmsg{
  color: red;
    text-align: left
}
.Logins{
    display :block;
    position: relative;
    background: #e6eef0f2;
    margin-top: 20px;
    border-radius: 10px;
    box-shadow: 7px 0px 1px 5px #e3e8ec;
    max-width: 100px;
}
@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}
.validationmsg{
  color: red!important;
  display: block;
  position: fixed;
}
.formlable{
  margin-top: 1rem;
}
.formbutton{
  margin-top: 1rem;
  margin-right:10px;
}
.search {
  margin-left: 400px;
    margin-top: 7px;
}

ul li {
  font-size: 15px;
}
.App-header {
  background-color: #282c34;
  min-height: 10vh;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  font-size: calc(10px + 2vmin);
  color: white;
  justify-content: center;
}
.slideshowSlider {
  white-space: nowrap;
  transition: ease 1000ms;
}
.slideshow {
  margin: 0 auto;
  overflow: hidden;
}
.slide {
  display: inline-block;
  height: 420px;
  width: 100%;
}
.textbox{
  box-sizing: content-box;
}
.gridstyle{
  margin-top: 30px;
  border: 1px solid;
}
.slideshowDots {
  text-align: center;
}
.slideshowDot {
  display: inline-block;
  height: 20px;
  width: 20px;
  border-radius: 50%;
  cursor: pointer;
  margin: 5px 5px 0px;
  background-color: #c4c4c4;
}
.slideshowDot.active {
  background-color: #534d57;
}
.logoimg{
  border-radius: 4px;
  padding:5px;
  width:75px;
  text-align: left;
}
.txt{
  position: fixed;
  margin-left: 80px;
  font-size: x-large;
}
.list {
  text-align: left;
  max-width: 750px;
  margin: auto;
}
.submit-form {
  align-items: center;
  font-family: serif;
  display: block;
}
.edit-form {
  max-width: 300px;
  margin: auto;
}
.App-link {
  color: #61dafb;
}
.formstyle{
    display: inline-block;
    position: relative;
    padding: 150px 500px 100px 100px;
    font-size: 20px;
    font-weight: bold;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
