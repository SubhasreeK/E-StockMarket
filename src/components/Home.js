import React from 'react';
import Navbar from './Navbar/NewNavbar';
class Home extends React.Component{
    
    render(){
        return(
            <div>
                 <div>
                    <Navbar/>
                 </div>
            
            </div>
        )
    }
}
export default Home;