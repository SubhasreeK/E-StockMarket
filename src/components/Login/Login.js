import React, { Component } from "react";
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";
import backgroundimg from '../../image/background.jpg';

import { withRouter } from 'react-router-dom';


class Login extends Component {

  constructor(props){
    super(props);

    this.state = {
      email : null,
      password : null,
      login : false,
      store : null,
      formErrors : {email:'',password:''},
      emailValid: false,
      passwordValid : false,
      formValid:false,
      loginStatus : ''
    }
    localStorage.clear();
  }
  handleClick(){
    this.setState({login:!this.login})
  }
  
  componentDidMount(){
    this.storecollect()
  }
  storecollect(){
    let store=JSON.parse(localStorage.getItem('login'));
    if(store && store.login){
      this.setState({login:true,store:store})
    }
  }
  handleUserInput (e) {
    const name = e.target.name;
    const value = e.target.value;
    this.setState({[name]:value},() => {this.validationField(name,value)});
    
  }

  login(){
    let found =false;
      fetch('http://localhost:8080/login',{
        method : "POST",
        body : JSON.stringify(this.state)
      }).then((response)=>{
        response.json().then(
          (result)=>{console.warn("result",result);
          localStorage.setItem('login',JSON.stringify({
            login:true,
            token:result.token
          }))
          this.storecollect();
      })
      found =true;
      if(found){
        this.setState({
          loginStatus : 'Logged In Successfully'
        });
        this.props.history.push('/home');
      }
      else{
        this.setState({
          loginStatus : 'Login Failed! Invalid Username/Password'
        });
      }
      }).catch(errors=> {
        this.setState({
          loginStatus : 'Server Down'
        });
      })
  }
 
  validationField(fieldName,value){
    let fieldValidErrors = this.state.formErrors;
    let emailValid = this.state.emailValid;
    let passwordValid = this.state.passwordValid;

     switch(fieldName){
       case 'email': 
           emailValid = value.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i);
           fieldValidErrors.email = emailValid ?'':'Enter correct email format';
           break;
      case  'password':
          passwordValid = value.length>= 6;
          fieldValidErrors.password = passwordValid ?'':'Enter atleast 6 character';
          break;
      default :
        break;
     }
    
     this.setState({formErrors : fieldValidErrors,
        emailValid : emailValid,
        passwordValid : passwordValid
     },this.validateForm);
    }
    validateForm(){
      this.setState({formValid: this.state.emailValid && this.state.passwordValid});
    }
 
  errorClass(error) {
    return(error.length === 0 ? '' : 'has-error');
 }
  render(){
  return (
  <div className="login" style={{backgroundImage:` url(${backgroundimg})`,backgroundRepeat:'no-repeat',backgroundSize:'cover'}}>
      {
       
        <Form className="formstyle">
        <Form.Group size="lg" controlId="email" className={`${this.errorClass(this.state.formErrors.email)}`}>
          <Form.Label className="formlable">Email</Form.Label>
          <Form.Control
            autoFocus
            type="email"
            name="email"
            onChange={(e) => {this.handleUserInput(e)}}
          />
          <Form.Text className="validationmsg">{this.state.formErrors.email}</Form.Text>
        </Form.Group>
        <Form.Group size="lg" controlId="password" className={`${this.errorClass(this.state.formErrors.password)}`}>
          <Form.Label className="formlable">Password</Form.Label>
          <Form.Control
            type="password"
            name="password"
            onChange={(e) => {this.handleUserInput(e)}}
          />
          <Form.Text className="validationmsg">{this.state.formErrors.password}</Form.Text>
        </Form.Group>
        <Button className ="button" onClick ={()=>{this.login()}} size="lg" disabled={!this.state.formValid}>
          Login
        </Button>
        <div>
        <p className="validationmsg">{this.state.loginStatus}</p>
        </div>
      </Form>
      
    
    
      
      }
     
    </div>
  );
  };
}
export default withRouter(Login);
