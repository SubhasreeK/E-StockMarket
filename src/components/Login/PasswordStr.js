import React from 'react';
 const PasswordStr = props =>{
     var strColor;
     var strWidth;
     switch(props.score){
         case 1:
             strColor ='red';
             strWidth ='20%';
             break;
         case 2:
             strColor ='orange';
             strWidth ='50%';
             break;
         case 3:
             strColor ='yellow';
             strWidth ='75%';
             break;
         case 4:
             strColor ='green';
             strWidth ='100%';
             break;
        default:
            strColor ='green';
             strWidth ='100%';
             break;
     }
     var style = {backgroundColor : strColor,height:'5px',width: strWidth, transition: "all 300ms ease-in-out"}
     return(
        <div>
         <p className='pwStrWeek'>week</p>
         <p className='pwStrStrong'>strong</p>
         <div style={style}/>
         </div>
     )
    }
 export default PasswordStr;