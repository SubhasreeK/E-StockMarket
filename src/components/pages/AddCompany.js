import React,{useState} from 'react';
import {useHistory} from 'react-router-dom';
import {useForm} from 'react-hook-form';
import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { DesktopDatePicker } from '@mui/x-date-pickers/DesktopDatePicker';
import {TextField,FormControl,InputLabel,Input} from '@material-ui/core';
import CompanyService from '../../services/ServiceCall';
import ErrorOutput from './FormValidation';

const AddCompany =() =>{
    let navigate = useHistory();
    const {handleSubmit} = useForm();

    const initialCompanyState = {
        id: null,
        companycode : null,
        companyname : "",
        stockprice : null,
        website: null,
        turnover:null,
        minvalue : null,
        maxvalue : null,
        avgvalue : null
    }
  
    const[newcomp, setNewcomp] = useState(initialCompanyState);
    const[submitted, setSubmitted] = useState(false);
    const handleInputChange = event =>{
        console.log("Handle Inputs : "+event.target);
        const {name,value}= event.target ;
        setNewcomp({...newcomp,[name]:value});
    };
    const saveCompany =()=>{
        var data ={
            companycode : newcomp.companycode,
            companyname : newcomp.companyname,
            stockprice : newcomp.stockprice,
            website :   newcomp.website,
            turnover :  newcomp.turnover,
            minvalue : newcomp.minvalue,
            maxvalue : newcomp.maxvalue,
            avgvalue : newcomp.avgvalue
        };
     
        console.log(JSON.stringify(data, null, 2));
    CompanyService.create(data)
    .then(response =>{
        setNewcomp({
            id : response.data.id,
            companycode : response.data.companycode,
            companyname : response.data.companyname,
            website : response.data.stockprice,
            startdate : response.data.website,
            turnover : response.data.turnover,
            minvalue : response.data.minvalue,
            maxvalue : response.data.maxvalue,
            avgvalue : response.data.avgvalue
        });
        setSubmitted(true);
        console.log(response.data);
    })
    .catch(e => {
        console.log(e);
    });
    };
    const newCompany =()=>{
        setNewcomp(initialCompanyState);
        setSubmitted(false);
    }
    const cancelCompany = () =>{
        navigate.push("/home");
   };
    return(
        <div className='submit-form '>
            {submitted ? (
                <div>
                    <h4>You Submitted Successfully!</h4>
                    <button className='btn btn-success' onClick={newCompany}>Add Company</button>
                </div>
            ):(
                <form  onSubmit={handleSubmit(saveCompany)}>
                <div className="container Logins">
                <h3>Add Company</h3>
               
                    <FormControl  variant='outlined' >
                        <InputLabel htmlFor='companyname'>Company Name</InputLabel>
                            <Input required id="companyname" className='form-control' onChange={handleInputChange} name="companyname"
                            value={newcomp.companyname} />
                        <ErrorOutput case={newcomp.companyname} name={'companyname'}/>
                    </FormControl> <br />
                   
                    <FormControl variant='outlined'>
                        <InputLabel htmlFor='companycode'>Company Code</InputLabel>
                        <Input required id="companycode" className='form-control' onChange={handleInputChange}
                        name="companycode" value={newcomp.companycode}/>
                    </FormControl> <br />
                    <FormControl  variant='outlined'>
                        <InputLabel htmlFor='stockprice'>Stock Price</InputLabel>
                        <Input required id="stockprice" className='form-control' onChange={handleInputChange} pattern="[0-9]*"
                        name="stockprice" value={newcomp.stockprice} inputProps={{ maxLength :10}}/>
                        <ErrorOutput case={newcomp.stockprice} name={'stockprice'}/>
                    </FormControl> <br />
                    <FormControl  variant='outlined'>
                        <InputLabel htmlFor='website'>Website</InputLabel>
                        <Input required id="website" className='form-control' onChange={handleInputChange} 
                        name="website" value={newcomp.website} inputProps={{ maxLength :50}}/>
                        <ErrorOutput case={newcomp.website} name={'website'}/>
                    </FormControl> <br />
                    <FormControl  variant='outlined'>
                        <InputLabel htmlFor='turnover'>Turnover</InputLabel>
                        <Input required id="turnover" className='form-control' onChange={handleInputChange} pattern="[0-9]*"
                        name="turnover" value={newcomp.turnover} inputProps={{ maxLength :10}}/>
                        <ErrorOutput case={newcomp.turnover} name={'turnover'}/>
                    </FormControl> <br />
                    
                   
                    <FormControl  variant='outlined'>
                        <InputLabel htmlFor='minvalue'>Minimum Value</InputLabel>
                        <Input required id="minvalue" className='form-control' onChange={handleInputChange}
                        name="minvalue" value={newcomp.minvalue} inputProps={{ maxLength :10}}/>
                         <ErrorOutput case={newcomp.minvalue} name={'minvalue'}/>
                    </FormControl> <br />
                    <FormControl  variant='outlined'>
                        <InputLabel htmlFor='maxvalue'>Maximum Value</InputLabel>
                        <Input required id="maxvalue" className='form-control' onChange={handleInputChange}
                        name="maxvalue" value={newcomp.maxvalue} inputProps={{ maxLength :10}}/>
                         <ErrorOutput case={newcomp.maxvalue} name={'maxvalue'}/>
                    </FormControl> <br />
                    <FormControl  variant='outlined'>
                        <InputLabel htmlFor='avgvalue'>Average Value</InputLabel>
                        <Input required id="avgvalue" className='form-control' onChange={handleInputChange}
                        name="avgvalue" value={newcomp.avgvalue} inputProps={{ maxLength :10}}/>
                         <ErrorOutput case={newcomp.avgvalue} name={'avgvalue'}/>
                    </FormControl> <br />

                
                <button type='submit' className="btn btn-success formbutton">Submit</button>
                <button onClick={cancelCompany} className="btn btn-success formlable">Cancel</button>
            </div>
            </form>
            )
            }
        </div>
    );
};

export default AddCompany;