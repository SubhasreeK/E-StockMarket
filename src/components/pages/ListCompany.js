import React , {useState,useEffect,useRef,useMemo} from 'react';
import {useTable} from 'react-table';
import {useHistory} from 'react-router-dom';
import CompanyService from '../../services/ServiceCall';
import {Row,Col} from 'react-bootstrap';
import {confirmAlert} from 'react-confirm-alert';
import 'react-confirm-alert/src/react-confirm-alert.css';
import axios from 'axios';
 const ListCompany =(props)=>{
     let history = useHistory();
     //const url = 'http://65.2.90.227:8000/api/v1.0/market/company/info/001';
    //const url1 = 'http://65.2.90.227:8000/api/v1.0/market/company/info/getall';
    //const url = 'http://localhost:8081/search';
    // const url1 = 'http://localhost:8082/list';
    const [companies,setCompanies] = useState([]);
    const [currentCompany,setCurrentCompany] = useState(null);
    const [currentIndex,setCurrentIndex] = useState(-1);
    const [searchCompany,setSearchCompany] = useState("");
    const companyRef =useRef();
    companyRef.current = companies;
    useEffect(()=>{
        retrieveCompany();
    },[]);
    const onChangesSearchCompany = e =>{
        const searchname = e.target.value;
        setSearchCompany(searchname);
    }
    const retrieveCompany = () =>{
       // axios.get(url1)
       CompanyService.getAll()
        .then(response => {
            setCompanies(response.data);
            console.log(JSON.stringify(response.data, null, 2));
        })
        .catch(e =>{
            console.log(e);
        });
    };
    const refreshList =()=>{
        retrieveCompany();
        setCurrentCompany(null);
        setCurrentIndex(-1);
    };
    const setActiveCompany = (company,index) =>{
        setCurrentCompany(company);
        setCurrentIndex(index);
    }
    const findbyname = () =>{
        //axios.get(url)
        const name = searchCompany; 
        console.log("Inside Find NAme = ",name); 
        CompanyService.get(name)
        .then(response =>{
            console.log(response.data);
            let resArr = [response.data];
            console.log(JSON.stringify(resArr, null, 2));
            setCompanies(resArr);
        })
        .catch(e => {
            console.log(e);
        });
    };
    const openCompany = (rowIndex) => {
        const id = companyRef.current[rowIndex].companyCode;
        //props.history.push("/viewcompany");
        //Yet To Work on it
        console.log("View Company = "+ id)
        history.push({
            pathname: '/home/viewcompany',
            state: { detail: id}
        });
      };
    const deleteConfirm =(rowIndex)=>{
        const id = companyRef.current[rowIndex].companyCode;
        const name = companyRef.current[rowIndex].companyName;
        console.log(id);
        confirmAlert({
            title:'Are you sure',
            message:`Do you want to delete the ${name}?`,
            buttons:[
                {
                    label:'Yes',
                    onClick:()=> deleteCompany(id)
                },
                {
                    label:'No'                }
                
            ]
        })
    }
    const deleteCompany = (id) => {

        //const id = companyRef.current[rowIndex].companycode;
        console.log("Inside Delete - "+ id);
        CompanyService.remove(id)
          .then((response) => {
            //props.history.push("/home");
            //history.push("/home");
            let newCompany = [...companyRef.current];
           // newCompany.splice(rowIndex, 1);
            setCompanies(newCompany);
            refreshList();
          })
          .catch((e) => {
            console.log(e);
          });
      };
      const columns = useMemo(
          ()=> [
        {
          Header : "Company Name",
          accessor : "companyName"
        },
        {
            Header : "Company Code",
            accessor : "companyCode"
        },
        {
            Header : "Company CEO",
            accessor : "companyCEO"
        },
        {
            Header : "Turn Over",
            accessor : "turnover"
        },
        {
            Header: "Actions",
            accessor: "actions",
            Cell : (props) =>{
                const rowIdx = props.row.id;
                return(
                    <div>
                        <span style={{marginRight:'10px'}} onClick={()=>openCompany(rowIdx)}>
                            <i className='far fa-edit action mr-2'></i>
                        </span>
                        <span onClick={()=> deleteConfirm(rowIdx)}>
                            <i className='fas fa-trash action'></i>
                        </span>
                    </div>
                );
            }
        }
    ],
    []
    );
    const {
        getTableProps,
        getTableBodyProps,
        headerGroups,
        rows,
        prepareRow,
      } = useTable({
        columns,
        data: companies,
      });
 return(
     <div className='list row container'>
         <div className='col-md-8'>
             <div className='input-group mb-3 formlable'>
                 <input type='text' className='form-control'
                 placeholder='Search by CompanyCode' value={searchCompany}
                 onChange={onChangesSearchCompany}/>

                 <div className='input-group-append'>
                     <button className='btn btn-outline-primary' type='button' onClick={findbyname}>Search</button>
                 </div>
             </div>
         </div>

         <div className='col-md-12 list'>
            <table className='table table-striped table-bordered tableStyle'
            {...getTableProps()}>
                <thead>
                    {headerGroups.map((headerGroup)=>(
                        <tr {...headerGroup.getHeaderGroupProps()}>
                            {headerGroup.headers.map((column)=>(
                                <th {...column.getHeaderProps()}>
                                    {column.render("Header")}
                                </th>
                            ))}
                        </tr>
                    ))}
                </thead>
                <tbody {...getTableBodyProps()}>
                    {rows.map((row,i)=>{
                        prepareRow(row);
                        return(
                            <tr {...row.getRowProps()}>
                                {row.cells.map((cell)=>{
                                    return(
                                        <td {...cell.getCellProps()}>{cell.render("Cell")}</td>
                                    );
                                })}
                            </tr>
                        )
                    })}
                </tbody>
            </table>
         </div>
       
     </div>
   
 )
}
export default ListCompany;