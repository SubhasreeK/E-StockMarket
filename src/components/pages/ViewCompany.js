import React,{useState,useEffect} from "react";
import {useParams, useHistory, useLocation} from 'react-router-dom';
import CompanyService from '../../services/ServiceCall';
import axios from 'axios';
const ViewCompany = props =>{
    //const url = 'http://43.205.53.10:8012/api/v1.0/market/stock/add';
    const {id} = useParams();
    const location = useLocation();
    let navigate = useHistory();
    const initialCompanyState = [{
        id:null,
        companyName:"",
        companyCode:"",
        price:null
    }
    ];
    
    const [currentCompany, setCurrentCompany] = useState(null);
    const [message,setMessage] = useState("");
    const getCompany = () =>{
        
         console.log("View - "+location.state.detail);
         const id = location.state.detail;
       CompanyService.get(id)
        .then(response => {
            setCurrentCompany(response.data);
            console.log({currentCompany});
        })
        .catch(e =>{
            console.log(e);
        });
    };
    useEffect(()=>{
            getCompany();
    },[]);
    const handleInputChange = event =>{
        const {name,value} = event.target;
        setCurrentCompany({...currentCompany,[name]:value});
    }
    const updateCompany= () =>{
        var data ={
            price : currentCompany.price,
        }
        console.log(data);
       axios.post(`http://43.205.53.10:8012/api/v1.0/market/stock/add/${currentCompany.companyCode}`,data)
        .then(response => {
            console.log(response.data);
            setMessage("Company has been Updated Sucsessfully!!")
        })
        .catch(e => {
            console.log(e);
            navigate.push("/home/listcompany"); 
        })
    }
    const cancelCompany = () =>{
         navigate.push("/home/listcompany");
    };
    return(
        <div>
            {currentCompany?(
                <div className="edit-form">
                    <h5>Edit Company Details</h5>
                    <form>
                       
                        <div className="form-group">
                            <label htmlFor="companyName">Company Name</label>
                            <input
                            type="text" disabled
                            className="form-control"
                            id="companyName" name="companyName" value={currentCompany.companyName} 
                            onChange={handleInputChange}></input>
                        </div>
                        <div className="form-group">
                            <label htmlFor="companyCode">Company Code</label>
                            <input
                            type="text" disabled
                            className="form-control" 
                            id="companyCode" name="companyCode" value={currentCompany.companyCode} 
                            onChange={handleInputChange}></input>
                        </div>
                        <div className="form-group">
                            <label htmlFor="price">Stock Price</label>
                            <input
                            type="text"
                            className="form-control"
                            id="price" name="price" value={currentCompany.price} 
                            onChange={handleInputChange}></input>
                        </div>
                     
                    </form>
                    <button type="submit" className="btn btn-success formbutton" onClick={updateCompany}>Update</button>
                    <button className="btn btn-warning formlable" onClick={cancelCompany}>Back</button>
                <p>{message}</p>
                </div>
            ):(
                <div>
                    <br/>
                    <p>Please click here..</p>
                </div>
            )}
        </div>
    );
};
export default ViewCompany;