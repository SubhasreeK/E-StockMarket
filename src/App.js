import React,{Component} from 'react';
import { Switch, Route } from 'react-router-dom';
import './App.css';
import "bootstrap/dist/css/bootstrap.min.css";
import "@fortawesome/fontawesome-free/css/all.css";
import "@fortawesome/fontawesome-free/js/all.js";
import Headers from './components/Header';
import Login from './components/Login/Login';
import Home from './components/Home';
import AutoLogoutTimer from './AutoLogout';

class App extends Component{

 
 render(){
  return (
    <div className="App">
     <Headers/>
      <Switch>
        <Route exact path='/'><Login/></Route>
        <Route path='/home'><AutoLogoutTimer ComposedClass={Home} /></Route>
      </Switch>
      
    </div>
  );
}
}


export default App;
