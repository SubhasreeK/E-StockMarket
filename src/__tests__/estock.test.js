import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import "@testing-library/jest-dom";
import {render, screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {Router} from 'react-router-dom'
import {createMemoryHistory} from 'history';
import Adapter from 'enzyme-adapter-react-16';
import Navbar from '../components/Navbar/NewNavbar';
import Login from '../components/Login/Login';
import Home from '../components/Home';
import AddCompany from '../components/pages/AddCompany';
import ListCompany from '../components/pages/ListCompany';
import ViewCompany from '../components/pages/ViewCompany';
import {App} from '../App';


Enzyme.configure({adapter: new Adapter()});
test('full app rendering', async()=>{
    const history = createMemoryHistory()
    render(
        <Router location={history.location} navigator={history}>
            <App/>
        </Router>,
    )
    const user = userEvent.setup();
    expect(screen.getByTitle(/E-Stock Market/i)).toBeInTheDocument();
    await user.click(screen.getByText(/Login/i))
    expect(screen.getElements(/NavBar/i)).toBeInTheDocument()

})
describe("Login",() => {
    it("should render Login Component",()=>{
        const wrapper = shallow(<Login/>);
        expect(wrapper.getElements()).toMatchSnapshot();
    });
});
describe("Home",() => {
    it("should render Home Component",()=>{
        const wrapper = shallow(<Home/>);
        expect(wrapper.getElements()).toMatchSnapshot();
    });
});
describe("AddCompany",() => {
    it("should render AddCompany Component",()=>{
        const wrapper = shallow(<AddCompany/>);
        expect(wrapper.getElements()).toMatchSnapshot();
    });
});
describe("ListCompany",() => {
    it("should render ListCompany Component",()=>{
        const wrapper = shallow(<ListCompany/>);
        expect(wrapper.getElements()).toMatchSnapshot();
    });
});
describe("ViewCompany",() => {
    it("should render ViewCompany Component",()=>{
        const wrapper = shallow(<ViewCompany/>);
        expect(wrapper.getElements()).toMatchSnapshot();
    });
});
describe("<Navbar/>",()=>{
    it("renders text input correctly",()=>{
        const navcomponent = shallow(<Navbar/>);
        const navLinks = navcomponent.find('Link');
        navLinks.simulate('click');
        const element= navcomponent.find('div div');
        expect (element.length).toBe(0);
        //expect(navcomponent.getElements()).toBeVisible();
    });
});
describe("<AddCompany/>",()=>{
    it("renders text input correctly",()=>{
        const addcomponent = shallow(<AddCompany/>);
        const form = addcomponent.find('input');
        form.props().onChange({target:{
            name: 'companyname',
            value : 'companyname'
        }})
        expect(addcomponent.state('input')).toBeDefined();
    });
});