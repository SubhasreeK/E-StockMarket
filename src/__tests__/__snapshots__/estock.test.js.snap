// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCompany should render AddCompany Component 1`] = `
Array [
  <div
    className="submit-form "
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="container Logins"
      >
        <h3>
          Add Company
        </h3>
        <WithStyles(ForwardRef(FormControl))
          fullWidth={true}
          variant="filled"
        >
          <WithStyles(ForwardRef(InputLabel))
            htmlFor="companyname"
          >
            Company Name
          </WithStyles(ForwardRef(InputLabel))>
          <WithStyles(ForwardRef(Input))
            className="form-control"
            id="companyname"
            name="companyname"
            onChange={[Function]}
            required={true}
            value=""
          />
          <ErrorOutput
            case=""
            name="companyname"
          />
        </WithStyles(ForwardRef(FormControl))>
        <WithStyles(ForwardRef(FormControl))
          fullWidth={true}
          variant="filled"
        >
          <WithStyles(ForwardRef(InputLabel))
            htmlFor="companycode"
          >
            Company Code
          </WithStyles(ForwardRef(InputLabel))>
          <WithStyles(ForwardRef(Input))
            className="form-control"
            id="companycode"
            name="companycode"
            onChange={[Function]}
            required={true}
            value={null}
          />
        </WithStyles(ForwardRef(FormControl))>
        <WithStyles(ForwardRef(FormControl))
          fullWidth={true}
          variant="filled"
        >
          <WithStyles(ForwardRef(InputLabel))
            htmlFor="stockprice"
          >
            Stock Price
          </WithStyles(ForwardRef(InputLabel))>
          <WithStyles(ForwardRef(Input))
            className="form-control"
            id="stockprice"
            inputProps={
              Object {
                "maxLength": 10,
              }
            }
            name="stockprice"
            onChange={[Function]}
            pattern="[0-9]*"
            required={true}
            value={null}
          />
          <ErrorOutput
            case={null}
            name="stockprice"
          />
        </WithStyles(ForwardRef(FormControl))>
        <LocalizationProvider
          dateAdapter={[Function]}
        >
          <div
            className="formlable"
          >
            <ForwardRef(DesktopDatePicker)
              label="Start Date : "
              maxDate={2100-01-01T00:00:00.000Z}
              minDate={2000-01-01T00:00:00.000Z}
              onChange={[Function]}
              renderInput={[Function]}
              required={true}
              value={null}
            />
          </div>
          <div
            className="formlable"
          >
            <ForwardRef(DesktopDatePicker)
              className="formlable"
              label="End Date : "
              maxDate={2100-01-01T00:00:00.000Z}
              minDate={2022-06-13T05:19:23.878Z}
              onChange={[Function]}
              renderInput={[Function]}
              required={true}
              value={null}
            />
          </div>
        </LocalizationProvider>
        <WithStyles(ForwardRef(FormControl))
          fullWidth={true}
          variant="filled"
        >
          <WithStyles(ForwardRef(InputLabel))
            htmlFor="minvalue"
          >
            Minimum Value
          </WithStyles(ForwardRef(InputLabel))>
          <WithStyles(ForwardRef(Input))
            className="form-control"
            id="minvalue"
            inputProps={
              Object {
                "maxLength": 10,
              }
            }
            name="minvalue"
            onChange={[Function]}
            required={true}
            value={null}
          />
          <ErrorOutput
            case={null}
            name="minvalue"
          />
        </WithStyles(ForwardRef(FormControl))>
        <WithStyles(ForwardRef(FormControl))
          fullWidth={true}
          variant="filled"
        >
          <WithStyles(ForwardRef(InputLabel))
            htmlFor="maxvalue"
          >
            Maximum Value
          </WithStyles(ForwardRef(InputLabel))>
          <WithStyles(ForwardRef(Input))
            className="form-control"
            id="maxvalue"
            inputProps={
              Object {
                "maxLength": 10,
              }
            }
            name="maxvalue"
            onChange={[Function]}
            required={true}
            value={null}
          />
          <ErrorOutput
            case={null}
            name="maxvalue"
          />
        </WithStyles(ForwardRef(FormControl))>
        <WithStyles(ForwardRef(FormControl))
          fullWidth={true}
          variant="filled"
        >
          <WithStyles(ForwardRef(InputLabel))
            htmlFor="avgvalue"
          >
            Average Value
          </WithStyles(ForwardRef(InputLabel))>
          <WithStyles(ForwardRef(Input))
            className="form-control"
            id="avgvalue"
            inputProps={
              Object {
                "maxLength": 10,
              }
            }
            name="avgvalue"
            onChange={[Function]}
            required={true}
            value={null}
          />
          <ErrorOutput
            case={null}
            name="avgvalue"
          />
        </WithStyles(ForwardRef(FormControl))>
        <button
          className="btn btn-success formbutton"
          type="submit"
        >
          Submit
        </button>
        <button
          className="btn btn-success formlable"
          onClick={[Function]}
        >
          Cancel
        </button>
      </div>
    </form>
  </div>,
]
`;

exports[`Home should render Home Component 1`] = `
Array [
  <div>
    <div>
      <Navbars />
    </div>
  </div>,
]
`;

exports[`ListCompany should render ListCompany Component 1`] = `
Array [
  <div
    className="list row container"
  >
    <div
      className="col-md-8"
    >
      <div
        className="input-group mb-3 formlable"
      >
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Search by CompanyName"
          type="text"
          value=""
        />
        <div
          className="input-group-append"
        >
          <button
            className="btn btn-outline-primary"
            onClick={[Function]}
            type="button"
          >
            Search
          </button>
        </div>
      </div>
    </div>
    <div
      className="col-md-12 list"
    >
      <table
        className="table table-striped table-bordered"
        role="table"
      >
        <thead>
          <tr
            role="row"
          >
            <th
              colSpan={1}
              role="columnheader"
            >
              Company Name
            </th>
            <th
              colSpan={1}
              role="columnheader"
            >
              Company Code
            </th>
            <th
              colSpan={1}
              role="columnheader"
            >
              Stock Price
            </th>
            <th
              colSpan={1}
              role="columnheader"
            >
              Maxmium
            </th>
            <th
              colSpan={1}
              role="columnheader"
            >
              Minimum
            </th>
            <th
              colSpan={1}
              role="columnheader"
            >
              Average
            </th>
            <th
              colSpan={1}
              role="columnheader"
            >
              Actions
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        />
      </table>
    </div>
  </div>,
]
`;

exports[`Login should render Login Component 1`] = `
Array [
  <Context.Consumer>
    [Function]
  </Context.Consumer>,
]
`;

exports[`ViewCompany should render ViewCompany Component 1`] = `
Array [
  <div>
    <div>
      <br />
      <p>
        Please click here..
      </p>
    </div>
  </div>,
]
`;
